code/modules/mining/lavaland/necropolis_chests.dm

```diff
@@ -1004,18 +976,22 @@
 	var/create_delay = 30
 	var/reset_cooldown = 50
 	var/timer = 0
-	var/static/list/banned_turfs = typecacheof(list(/turf/open/space/transit, /turf/closed))
+	var/static/list/banned_turfs = typecacheof(list(/turf/closed))
+	var/static/list/allowed_areas = list(/area/lavaland/surface/outdoors)
 
 /obj/item/lava_staff/afterattack(atom/target, mob/user, proximity_flag, click_parameters)
 	. = ..()
+	if(user.z != 5)
+		to_chat(user, "<span class='warning'>The staff's power is too dim to function this far from the necropolis")
+		return
```

Comment:
This makes it basically useless, you can already create lava on lavaland with an RCD.
Also don't check z level by comparing it to some abritary number, Z-Levels have no defined order as to what they are, so 5 could be anything and not just lavaland. Use the mapping helper defines.

Reply:
> with an RCD.

Which miners should generally not have easy access to. It also destroys lava. 

> Use the mapping helper defines.

Going to need to see an example of this being used because I could not get them to work. Additionally, to the best I could tell lavaland is actually always 5. Other items which are only supposed to work on lavaland also specifically reference z level 5 instead of using helper defines.